megatron/core/pipeline_parallel/p2p_communication.py

```diff
@@ -195,152 +229,116 @@ def _p2p_ops(*,
              tensor_recv_prev: Optional[torch.Tensor],
              tensor_send_next: Optional[torch.Tensor],
              tensor_recv_next: Optional[torch.Tensor],
+             within_component: bool = False,
              group: torch.distributed.ProcessGroup):
     # print("Using p2pOps without components")
     reqs = []
-    rank = get_pipeline_model_parallel_rank()
-    if get_pipeline_model_parallel_rank() % 2 == 0:
-        if tensor_send_next is not None:
-            send_next_req = torch.distributed.isend(
-                tensor=tensor_send_next,
-                dst=get_pipeline_model_parallel_next_rank(),
-                group=group,
-            )
-            reqs.append(send_next_req)
-
-        if tensor_recv_prev is not None:
-            recv_prev_req = torch.distributed.irecv(
-                tensor=tensor_recv_prev,
-                src=get_pipeline_model_parallel_prev_rank(),
-                group=group,
-            )
-            reqs.append(recv_prev_req)
-
-        if tensor_send_prev is not None:
-            send_prev_req = torch.distributed.isend(
-                tensor=tensor_send_prev,
-                dst=get_pipeline_model_parallel_prev_rank(),
-                group=group,
-            )
-            reqs.append(send_prev_req)
-
-        if tensor_recv_next is not None:
-            recv_next_req = torch.distributed.irecv(
-                tensor=tensor_recv_next,
-                src=get_pipeline_model_parallel_next_rank(),
-                group=group,
-            )
-            reqs.append(recv_next_req)
-
-    else:
-        if tensor_recv_prev is not None:
-            recv_prev_req = torch.distributed.irecv(
-                tensor=tensor_recv_prev,
-                src=get_pipeline_model_parallel_prev_rank(),
-                group=group,
-            )
-            reqs.append(recv_prev_req)
+    if tensor_send_next is not None:
+        send_next_req = torch.distributed.isend(
```

Comment:
The original code separate the case for even ranks and odd ranks, so they have matched the order of the send and receive (https://github.com/crankshaw-google/Megatron-LM/blob/325ddb2e4d0c31c75397fb3f4363d0a13f43a51c/megatron/core/pipeline_parallel/p2p_communication.py#L155)

Doesn't it needed here?

Reply:
Good point, I must have accidentally removed this, will add back.